vscode/src/client/analyzerClient.ts

```diff
@@ -109,21 +110,23 @@ export class AnalyzerClient {
     }
 
     this.outputChannel.appendLine(`Starting the kai rpc server ...`);
-    this.fireStateChange("starting");
+    this.fireServerStateChange("starting");
+
     try {
       const [kaiRpcServer, pid] = await this.startProcessAndLogStderr();
 
       kaiRpcServer.on("exit", (code, signal) => {
         this.outputChannel.appendLine(`kai rpc server exited [signal: ${signal}, code: ${code}]`);
-        this.fireStateChange("stopped");
+        this.fireServerStateChange("stopped");
       });
 
       this.kaiRpcServer = kaiRpcServer;
       this.outputChannel.appendLine(`kai rpc server successfully started [pid: ${pid}]`);
-      this.fireStateChange("readyToInitialize");
+      //TODO: Handle the case where the server is not ready to initialize
+      // this.fireServerStateChange("readyToInitialize");
```

Comment:
/Have you seen the server state change to readyToInitialize in error?  `startProcessAndLogStderr()` should not successfully return until the server reports ready.  Is it logging "waited 10000ms for the kai rpc server to be ready, continuing anyway"?

If yes, then this timeout block may need some extra work:
https://github.com/konveyor/editor-extensions/blob/8eb892794747dec1f8bda81bd3b18539895e3614/vscode/src/client/analyzerClient.ts#L203-L206

At the very least, let's move this TODO comment to that location.